src/components/common/Map/EnhancedMarker/index.js

```diff
@@ -0,0 +1,33 @@
+import React, { useState, useEffect } from 'react';
+import { arrayOf, object } from 'prop-types';
+import { Marker, Popup } from 'react-leaflet';
+import getTopicIcon from '../Icons';
+
+const initialTopicState = {
+  id: '',
+  label: ''
+};
+
+const EnhancedMarker = ({ target, topics }) => {
+  const [targetTopic, setTargetTopic] = useState(initialTopicState);
+
+  useEffect(() => {
+    if (topics.length != 0) {
+      const temp = topics.find(t => t.id == target.topicId);
+      setTargetTopic(temp);
+    }
+  }, [topics, target, setTargetTopic]);
```

Comment:
How about moving this to a redux selector? something like `getTargetTopic`, which does the `topics.find(...)`

Reply:
That would be nice, but as the access to redux is done by the useSelector hook, it would end up on something like:

`const targetTopic = useSelector(state =>
    state.targetReducer.topics.find(t => t.id == target.topicId));`

isn't it? I don't know if this is a better approach or not 😅